stl/inc/chrono

```diff
@@ -5428,113 +5466,67 @@ struct formatter<_CHRONO day, _CharT> {
         // [time.format]/6
         if (_List.empty()) {
             _No_chrono_specs = true;
-            _Conv_spec       = _Chrono_specs<_CharT>{._Type = _CharT{'d'}};
-            _Specs._Chrono_specs_list.push_back(_Conv_spec);
-            _Stored_specs = _Specs;
-            return _Res_iter;
-        }
-
-        auto _Target_conv_spec_iter =
-            _STD find_if(_List.begin(), _List.end(), [](const auto& _Cs) { return _Cs._Type != '\0'; });
-
-        if (_Target_conv_spec_iter == _List.end()) { // no conversion-spec, but has literal chars
-            _No_chrono_specs = true;
-            _Conv_spec       = _Chrono_specs<_CharT>{._Type = _CharT{'d'}};
-            _Specs._Chrono_specs_list.push_back(_Conv_spec);
-            _Stored_specs = _Specs;
             return _Res_iter;
         }
 
-        _Chrono_specs<_CharT> _Target_conv_spec = *_Target_conv_spec_iter;
-        _Conv_spec                              = _Target_conv_spec;
-        // only 'd' and 'e' acceptable types for day
-        if (_Target_conv_spec._Type != 'd' && _Target_conv_spec._Type != 'e') {
-            _THROW(format_error("Invalid type for type day."));
-        }
-
-        if (_Target_conv_spec._Modifier != '\0' && _Target_conv_spec._Modifier != 'O') {
-            _THROW(format_error("Invalid modifier for type day."));
-        }
-
-        _Target_conv_spec_iter =
-            _STD find_if(_Target_conv_spec_iter + 1, _List.end(), [](const auto& _Cs) { return _Cs._Type != '\0'; });
-
-        if (_Target_conv_spec_iter != _List.end()) {
-            _THROW(format_error("day requires one conversion-spec."));
+        for (const auto& _Spec : _List) {
+            if (_Spec._Type != '\0' && _Spec._Type != 'd' && _Spec._Type != 'e') {
+                _THROW(format_error("Invalid type for type day."));
+            }
+            if (_Spec._Modifier != '\0' && _Spec._Modifier != 'O') {
+                _THROW(format_error("Invalid modifier for type day."));
+            }
         }
 
-        _Stored_specs = _Specs;
         return _Res_iter;
     }
 
     template <class _FormatContext>
     typename _FormatContext::iterator format(const _CHRONO day& _Val, _FormatContext& _FormatCtx) {
         using _Ctx_iter = typename _FormatContext::iterator;
 
-        const basic_string<_CharT> _Invalid_msg = _STATICALLY_WIDEN(_CharT, " is not a valid day");
         basic_stringstream<_CharT> _Stream;
-        _Stream.imbue(_FormatCtx.locale());
-        basic_string<_CharT> _Str;
-        int _Num_char = 2;
-        if (_No_chrono_specs && !_Val.ok()) {
-            _Stream << static_cast<unsigned int>(_Val);
+
+        if (_No_chrono_specs) {
             if (static_cast<unsigned int>(_Val) < 10) { // _Val == 0

```

Comment:
Should the `// _Val == 0` comment be dropped now? (It looks like it was a todo/fixme.) I see that `%d` specifies "If the result is a single decimal digit, it is prefixed with 0." and `day`'s `operator<<` says to use `%d` for not-OK values, so (if I'm reading this correctly) `"00 is not a valid day"` is expected.